Firestore/core/src/firebase/firestore/local/leveldb_remote_document_cache.mm

```diff
@@ -92,17 +92,30 @@
 DocumentMap LevelDbRemoteDocumentCache::GetMatching(FSTQuery* query) {
   DocumentMap results;
 
+  // Use the query path as a prefix for testing if a document matches the query.
+  const model::ResourcePath& query_path = query.path;
+  unsigned long immediate_children_path_length = query_path.size() + 1;
```

Comment:
Please don't use types based on `long` unless you're interoperating with a specific system API that requires it. This should be `size_t`.